protocols/virtualfund/virtualfund.go

```diff
@@ -657,3 +648,116 @@ func ConstructObjectiveFromMessage(m protocols.Message, myAddress types.Address,
+		return protocols.SideEffects{}, errors.New("only the proposer can propose a ledger update")
+	}
+
+	sideEffects := protocols.SideEffects{}
+
+	supported, err := ledger.LatestSupportedState()
+	if err != nil {
+		return protocols.SideEffects{}, fmt.Errorf("error finding a supported state: %w", err)
+	}
+
+	// Clone the state and update the turn to the next turn.
+	nextState := supported.Clone()
+	nextState.TurnNum = nextState.TurnNum + 1
+	// Update the outcome with the guarantee.
+	nextState.Outcome, err = nextState.Outcome.DivertToGuarantee(left, right, leftAmount, rightAmount, o.V.Id)
+	if err != nil {
+		return protocols.SideEffects{}, fmt.Errorf("error updating ledger channel outcome: %w", err)
+	}
+
+	// Sign the state and add it to the ledger.
+	ss, err := ledger.SignAndAddState(nextState, sk)
+	if err != nil {
+		return protocols.SideEffects{}, fmt.Errorf("error adding signed state: %w", err)
+	}
+
+	// Add a message with the signed state
+	messages := protocols.CreateSignedStateMessages(o.Id(), ss, ledger.MyIndex)
+	sideEffects.MessagesToSend = append(sideEffects.MessagesToSend, messages...)
+
+	return sideEffects, nil
+}
+
+// acceptLedgerUpdate checks for a ledger state proposal and accepts that proposal if it satisfies the expected guarantee.
+func (o *Objective) acceptLedgerUpdate(ledgerConnection Connection, sk *[]byte) (protocols.SideEffects, error) {
+	ledger := ledgerConnection.Channel
+	proposed, ok := ledger.Proposed()
+
+	if !ok {
+		return protocols.SideEffects{}, fmt.Errorf("no proposed state found for ledger channel %s", ledger.Id)
+	}
+
+	supported, err := ledger.LatestSupportedState()
+	if err != nil {
+		return protocols.SideEffects{}, fmt.Errorf("no supported state found for ledger channel: %w", err)
+	}
+
+	// Determine if we are left or right in the guarantee and determine our amounts.
+	ourAddress := types.AddressToDestination(ledger.Participants[ledger.MyIndex])
+	ourDeposit := ledgerConnection.GuaranteeInfo.LeftAmount
+	if ledger.MyIndex == 1 {
+		ourDeposit = ledgerConnection.GuaranteeInfo.RightAmount
+	}
+
+	ourPreviousTotal := supported.Outcome.TotalAllocatedFor(ourAddress)
+	ourNewTotal := proposed.Outcome.TotalAllocatedFor(ourAddress)
+
+	// Our new total should just be our previous total minus our deposit.
+	proposedMaintainsOurFunds := ourNewTotal.Add(ourDeposit).Equal(ourPreviousTotal)
+
+	proposedAffordsGuarantee := proposed.Outcome.Affords(ledgerConnection.ExpectedGuarantees, ledger.OnChainFunding)
+
+	if proposedMaintainsOurFunds && proposedAffordsGuarantee {
+
+		ss, err := ledger.SignAndAddState(proposed, sk)
+		if err != nil {
+			return protocols.SideEffects{}, fmt.Errorf("error adding signed state: %w", err)
+		}
+		sideEffects := protocols.SideEffects{}
+		messages := protocols.CreateSignedStateMessages(o.Id(), ss, ledger.MyIndex)
+		sideEffects.MessagesToSend = append(sideEffects.MessagesToSend, messages...)
+		return sideEffects, nil
+	}
+
+	return protocols.SideEffects{}, nil
```

Comment:
I think it makes the ledger protocol slightly more robust if we just ignore updates that were not interested in instead of explicitly rejecting them.

This means we don't have to worry about updates to the ledger channel outside of the objective causing us to return a rejection message. We simply wait until we a ledger update that satisfies our guarantees before approving.